portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,20 +30,47 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> comments;
+  class Comment {
+      private String name;
+      private String comment; 
+      private long timestamp;
 
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
+      public Comment(String name, String comment, long timestamp) {
+          this.name = name;
+          this.comment = comment; 
+          this.timestamp = timestamp;
+      }
   }
 
+  private List<Comment> comments;
+
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    comments = new ArrayList<>();
+    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int numComments = getLimit(request);
+
+    for (Entity entity : results.asIterable()) {
+      if (comments.size() == numComments) break;
+
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
Hey, feel free to reach out to me directly if you have questions. Should this be in this PR? I was under the impression it was only about the placeholder chart.

Reply:
This is from an old PR, and I am not sure why it is included in this PR. The only file I changed was the script.js file to include the chart... I am not entirely sure what to do.

Reply:
Great, thank you for the fix.